Sources/MongoSwift/BSON/DocumentIterator.swift

```diff
@@ -161,16 +166,34 @@ public class DocumentIterator: IteratorProtocol {
     }
 
     /// Internal helper function for explicitly accessing the `bson_iter_t` as an unsafe pointer
+#if compiler(>=5.0)
```

Comment:
I think I would find the differences between this function across compiler versions more readable if we could isolate the 1-2 different lines within the if/else rather than redefining the entire thing. I think you did this a couple places above.  something like
```swift
internal func withBSONIterPointer<Result>(_ body: (BSONIterPointer) throws -> Result) rethrows -> Result {
    return try withUnsafePointer(to: self.iter) { iterPtr in
#if compiler(>=5.0)
        try body(OpaquePointer(iterPtr))
#else 
        try body(iterPtr)
#endif
    }
}
```
makes it really clear that all that's changing is whether we cast to `OpaquePointer`.

Reply:
wow that's weird.. this must have been a result of fighting the compiler, I swear they used to have different signatures 🤔

Reply:
lol I felt like I was doing one of those "spot the differences between these two pictures" game 🕵️‍♀️